### PR TITLE
[IMP] crm: no auto commit

### PR DESCRIPTION
**Pull request by nseinlet**
commit at each lead makes the system spend 80% of the time recomputing bayesian probabilities. Avoid this commit speed up the process while not altering the final result.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by tfr-odoo** [comment]
I would continue to commit to limit the risk of concurrent update, without commit we will have a huge transaction of several minutes. We have to find the sweat spot between the time for the transaction and the size. Maybe it's around 100-200

**Comment by tde-banana-odoo** [comment]
> I would continue to commit to limit the risk of concurrent update, without commit we will have a huge transaction of several minutes. We have to find the sweat spot between the time for the transaction and the size. Maybe it's around 100-200

Agreed. Notably if you have an issue it would lead to a rollback and then back again on the same leads ... if we commit each 50/100 leads, we already remove most of time lost in recomputation.

**Comment by nseinlet** [comment]
image

File: addons/crm/models/crm_team_member.py [hunk 1]
```diff
@@ -174,9 +176,7 @@ def _assign_and_convert_leads(self, work_days=2):
                     user_ids=member_data['team_member'].user_id.ids
                 )
 
-                # auto-commit except in testing mode
-                auto_commit = not getattr(threading.currentThread(), 'testing', False)
-                if auto_commit:
+                if auto_commit and counter % 100 == 0:
```
Comment (tde-banana-odoo) [inline addons/crm/models/crm_team_member.py:179]: Couldn't we allow its tuning through a config parameter (using 100 as default value if not set) ? We already allow crm.assignment.delay and crm.assignment.bundle, we could have crm.assignment.commit.bundle . That way if we have to tweak it on some db it is easy to do.

**Review by tde-banana-odoo — COMMENTED** [review]
Riviou !

**Comment by tde-banana-odoo** [comment]
Could you also link task ID-2475352 in commit message ? I created it so that we have a trace in le Odoo.

**Comment by tde-banana-odoo** [comment]
If I may be the one citing guidelines, pour une fois: "commit header should make a valid sentence if concatenated with 'if applied, this commit will <header>". This does not work with current commit header "less auto commit". Probably to update with something like "improve assign performance by committing less often" .

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**